pull.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @module  audio-generator
+ */
+'use strict';
+
+var Generator = require('./index');
+
+module.exports = function Stream (fn, opts) {
+	let generate = Generator(fn, opts);
+	let ended = false;
+
+	let stream = (end, cb) => {
+		if (end || ended) {
+			ended = true;
+			return cb && cb(end);
+		}
+
+		return cb(null, generate());
+	}
+
+	stream.abort = () => {ended = true}
```

Comment:
i think the reason is that if you have a pipeline from source to sink and _abort with `sink.abort(err, cb?)`, the abort signal will propagate synchronously to the source automatically, so i think `.abort` is only most often found on either sink streams or streams with an underlying resource (a file descriptor, network connection, etc).